src/pages/Article.js

```diff
@@ -1,11 +1,16 @@
 // @flow
 
+import { Subscribe } from 'unstated';
+
 import Header from 'components/Header';
+import App from 'lib/AppContainer';
 
-const Article = ({ data }) => (
+const Article = () => (
     <>
-        <Header data={data} />
-        <h1>{data.config.page.headline}</h1>
+        <Header />
+        <Subscribe to={[App]}>
+            {app => <h1>{app.state.config.page.headline}</h1>}
```

Comment:
we need to think about the structure a bit! `app.state.config.page.headline` is a bit unweidly

Reply:
and flow-type it probably...

Reply:
Is it possible to inject multiple `Container`s? Maybe splitting up our app state would make this less cumbersome.

Reply:
ooh that's an interesting idea. yeah totally possible. would it make it hard to know what's where i wonder?

Reply:
I think splitting `state` used for interactivity and `state` that stores article data and page config would make sense? Eg. It doesn't feel right storing state to indicate whether a menu is open or closed alongside article data.

Reply:
Yeah, that's a trade-off. In this instance, I can't quite see, from a content point of view, where `config` ends and `header` begins.

But I agree, it's not up to the component to know where on the `AppContainer` to find the data it's looking for. It sounds like this belongs in some `mapContainerToProps` function which plucks out the data this component needs. Or a `reducer`.

Reply:
Like:

```js
function mapStateToProps(app) {
  return { headline: app.state.config.page.headline }
}

const Article = ({headline}) => {
  <Header>
    <h1>{headline}</h1>
  </Header>
}

export default connect(mapStateToProps)(Article)
```

Reply:
Thinking about this state related to interactivity should sit at a component level so you'd still use `this.setState` and `this.state` at that level, and use `Unstate` for accessing global state (page config and article data)

Reply:
yeah i think that's how i saw it? component like the nav can contain their own state (expanded), but components within it might also want to respond to global state changes (pulling user data out of cookie)